fluent-plugin-enhance-k8s-metadata/lib/fluent/plugin/filter_enhance_k8s_metadata.rb

```diff
@@ -66,23 +66,22 @@ def decorate_record(record)
         @in_namespace_ac.each { |ac| namespace_name ||= ac.call(record) }
         @in_pod_ac.each { |ac| pod_name ||= ac.call(record) }
         if namespace_name.nil?
-          log.debug "Record doesn't have [#{@in_namespace_path}] field"
+          log.trace "Record doesn't have [#{@in_namespace_path}] field"
         elsif pod_name.nil?
-          log.debug "Record doesn't have [#{@in_pod_path}] field"
+          log.trace "Record doesn't have [#{@in_pod_path}] field"
         else
           if record.key? 'service'
             record['prometheus_service'] = record['service']
             record.delete('service')
           end
           metadata = get_pod_metadata(namespace_name, pod_name)
-          # service = @pods_to_services[pod_name]
-          # metadata['service'] = {'service' => service.sort!.join('_')} if !(service.nil? || service.empty?)
+          service = @pods_to_services[pod_name]
```

Comment:
Is it possible we also use a ThreadSafe cache? Not sure if it helps though.

Reply:
I'm not sure if it helps, it's definitely something to consider - but I think that can be part of a separate PR after we verify that it works

Reply:
sg